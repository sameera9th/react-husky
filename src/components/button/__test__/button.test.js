import React from 'react'
import ReactDOM from 'react-dom'
import Button from '../index'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'

it('rendered without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Button />, div)
})

it('renders button correctly', () => {
  const { getByTestId } = render(<Button label="Click me please" />)
  expect(getByTestId('button')).toHaveTextContent('Click me please')
})

it('renders button correctly', () => {
  const { getByTestId } = render(<Button label="Save" />)
  expect(getByTestId('button')).toHaveTextContent('Save')
})

it('matches snapshot save', () => {
  const tree = renderer.create(<Button label="Save" />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('matches snapshot cancel', () => {
  const tree = renderer.create(<Button label="Cancel" />).toJSON()
  expect(tree).toMatchSnapshot()
})
