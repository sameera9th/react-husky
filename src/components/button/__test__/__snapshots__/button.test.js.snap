// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot cancel 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Cancel
</div>
`;

exports[`matches snapshot save 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Save
</div>
`;
